core/src/main/java/jenkins/security/SecurityListener.java

```diff
@@ -69,13 +69,31 @@ protected void failedToAuthenticate(@Nonnull String username){}
      */
     protected void loggedIn(@Nonnull String username){}
 
+    /**
+     * @since TODO
+     *
+     * Fired when a new user account has been created.
+     *
+     * @param username the user
+     */
+    protected void userCreated(@Nonnull String username) {}
+
     /**
      * Fired when a user has failed to log in.
      * Would be called after {@link #failedToAuthenticate}.
      * @param username the user
      */
     protected void failedToLogIn(@Nonnull String username){}
 
+    /**
+     * @since TODO
+     *
+     * Fired when new user account creation fails.
+     *
+     * @param username the user
+     */
+    protected void failedToCreateUser(@Nonnull String username){}
```

Comment:
This signature does not look particularly useful without a cause (and note the possibility of 0…N+1 causes).

In fact, this seems to be lacking a use case. What's the benefit of knowing that someone didn't enter an email address?

Reply:
as discussed earlier, I'll be pulling this method back. Will reintroduce it later if we find a more valid use case for it